source/common/protobuf/utility.h

```diff
@@ -219,6 +219,35 @@ class ProtoExceptionUtil {
                                             const Protobuf::Message& message);
 };
 
+// Move semantics friendly StatusOr that will used to replace ProtoBufUtil::StatusOr temporary to
+// achieve better performance.
+template <typename T, typename Status = ProtobufUtil::Status> class StatusOr {
```

Comment:
Let's not put a new definition of StatusOr in the Envoy namespace. You could call this ProtoStatusOr or something.

Reply:
Get it. Thanks.